From foo@baz Wed May 12 04:02:46 PM CEST 2021
From: Quentin Perret <qperret@google.com>
Date: Wed, 12 May 2021 12:28:52 +0000
Subject: Revert "of/fdt: Make sure no-map does not remove already reserved regions"
To: stable@vger.kernel.org
Cc: alexandre.torgue@foss.st.com, robh+dt@kernel.org, f.fainelli@gmail.com, ardb@kernel.org, gregkh@linuxfoundation.org, sashal@kernel.org
Message-ID: <20210512122853.3243417-2-qperret@google.com>

From: Quentin Perret <qperret@google.com>

This reverts commit 3cbd3038c9155038020560729cde50588311105d.
It is not really a fix, and the backport misses dependencies, which
breaks existing platforms.

Reported-by: Alexandre TORGUE <alexandre.torgue@foss.st.com>
Signed-off-by: Quentin Perret <qperret@google.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---
 drivers/of/fdt.c |   10 +---------
 1 file changed, 1 insertion(+), 9 deletions(-)

--- a/drivers/of/fdt.c
+++ b/drivers/of/fdt.c
@@ -1153,16 +1153,8 @@ int __init __weak early_init_dt_mark_hot
 int __init __weak early_init_dt_reserve_memory_arch(phys_addr_t base,
 					phys_addr_t size, bool nomap)
 {
-	if (nomap) {
-		/*
-		 * If the memory is already reserved (by another region), we
-		 * should not allow it to be marked nomap.
-		 */
-		if (memblock_is_region_reserved(base, size))
-			return -EBUSY;
-
+	if (nomap)
 		return memblock_mark_nomap(base, size);
-	}
 	return memblock_reserve(base, size);
 }
 
